--- gaim-2.0.0beta4/gtk/gtkblist.c	2006-10-18 10:37:48.000000000 -0700
+++ gaim-2.0.0beta4_accel/gtk/gtkblist.c	2006-11-15 00:41:48.000000000 -0800
@@ -5932,6 +5932,7 @@
 gaim_gtk_blist_update_accounts_menu(void)
 {
 	GtkWidget *menuitem = NULL, *submenu = NULL;
+	GtkAccelGroup *accel_group = NULL;
 	GList *l = NULL, *accounts = NULL;
 	gboolean disabled_accounts = FALSE;
 
@@ -5949,6 +5950,7 @@
 
 	for (accounts = gaim_accounts_get_all(); accounts; accounts = accounts->next) {
 		char *buf = NULL;
+		char *accel_path_buf = NULL;
 		GtkWidget *image = NULL;
 		GaimConnection *gc = NULL;
 		GaimAccount *account = NULL;
@@ -5956,11 +5958,13 @@
 		GdkPixbuf *pixbuf = NULL;
 
 		account = accounts->data;
+		accel_group = gtk_menu_get_accel_group(GTK_MENU(accountmenu));
 
 		if(gaim_account_get_enabled(account, GAIM_GTK_UI)) {
 			buf = g_strconcat(gaim_account_get_username(account), " (",
 					gaim_account_get_protocol_name(account), ")", NULL);
 			menuitem = gtk_image_menu_item_new_with_label(buf);
+			accel_path_buf = g_strconcat(N_("<GaimMain>/Accounts/"), buf, NULL);
 			g_free(buf);
 			status = gaim_account_get_active_status(account);
 			pixbuf = gaim_gtk_create_prpl_icon_with_status(account, gaim_status_get_type(status), 0.5);
@@ -5978,6 +5982,9 @@
 			gtk_widget_show(menuitem);
 
 			submenu = gtk_menu_new();
+			gtk_menu_set_accel_group(GTK_MENU(submenu), accel_group);
+			gtk_menu_set_accel_path(GTK_MENU(submenu), accel_path_buf);
+			g_free(accel_path_buf);
 			gtk_menu_item_set_submenu(GTK_MENU_ITEM(menuitem), submenu);
 			gtk_widget_show(submenu);
 
@@ -6046,6 +6053,8 @@
 		gtk_widget_show(menuitem);
 
 		submenu = gtk_menu_new();
+		gtk_menu_set_accel_group(GTK_MENU(submenu), accel_group);
+		gtk_menu_set_accel_path(GTK_MENU(submenu), N_("<GaimMain>/Accounts/Enable Account"));
 		gtk_menu_item_set_submenu(GTK_MENU_ITEM(menuitem), submenu);
 		gtk_widget_show(submenu);
 
