*** java-openid-sxip-0.9.1.74/src/net/openid/consumer/ConsumerManager.java	Tue Sep 18 14:41:35 2007
--- java-openid-sxip-0.9.1.74.new/src/net/openid/consumer/ConsumerManager.java	Fri Oct  5 17:43:23 2007
***************
*** 1409,1418 ****
                      discovered.getClaimedIdentifier();
  
              if ( assertId.equals(opSpecific) && discovered.isVersion2() &&
!                     discovered.getIdpEndpoint().equals(respEndpoint))
                  return discovered;
          }
  
          // stateless, bare response, or the user changed the ID at the OP
          DiscoveryInformation firstServiceMatch = null;
  
--- 1438,1447 ----
                      discovered.getClaimedIdentifier();
  
              if ( assertId.equals(opSpecific) && discovered.isVersion2() &&
!                     discovered.getIdpEndpoint().toString().equals(respEndpoint))
                  return discovered;
          }
  
          // stateless, bare response, or the user changed the ID at the OP
          DiscoveryInformation firstServiceMatch = null;
  
***************
*** 1427,1443 ****
          {
              DiscoveryInformation service = (DiscoveryInformation) iter.next();
  
              if (DiscoveryInformation.OPENID2_OP.equals(service.getVersion()))
                  continue;
  
              Identifier opSpecific = service.hasDelegateIdentifier() ?
                      service.getDelegateIdentifier() :
                      service.getClaimedIdentifier();
  
              if ( ! assertId.equals(opSpecific) || ! service.isVersion2() ||
!                     ! service.getIdpEndpoint().equals(respEndpoint) )
                  continue;
  
              // keep the first endpoint that matches
              if (firstServiceMatch == null) firstServiceMatch = service;
  
--- 1458,1474 ----
          {
              DiscoveryInformation service = (DiscoveryInformation) iter.next();
  
              if (DiscoveryInformation.OPENID2_OP.equals(service.getVersion()))
                  continue;
  
              Identifier opSpecific = service.hasDelegateIdentifier() ?
                      service.getDelegateIdentifier() :
                      service.getClaimedIdentifier();
  
              if ( ! assertId.equals(opSpecific) || ! service.isVersion2() ||
!                  ! service.getIdpEndpoint().toString().equals(respEndpoint) )
                  continue;
  
              // keep the first endpoint that matches
              if (firstServiceMatch == null) firstServiceMatch = service;
  
