diff -uNr -x .svn thematic-orig/functions.php thematic-comment-meta-filter/functions.php
--- thematic-orig/functions.php	2010-05-30 18:54:20.794664831 -0700
+++ thematic-comment-meta-filter/functions.php	2010-05-30 19:24:26.485333958 -0700
@@ -57,6 +57,9 @@
 // Load custom Comments filters
 require_once(THEMELIB . '/extensions/comments-extensions.php');
 
+// Load custom discussion filters
+require_once(THEMELIB . '/extensions/discussion-extensions.php');
+
 // Load custom Widgets
 require_once(THEMELIB . '/extensions/widgets-extensions.php');
 
diff -uNr -x .svn thematic-orig/library/extensions/discussion-extensions.php thematic-comment-meta-filter/library/extensions/discussion-extensions.php
--- thematic-orig/library/extensions/discussion-extensions.php	1969-12-31 16:00:00.000000000 -0800
+++ thematic-comment-meta-filter/library/extensions/discussion-extensions.php	2010-05-30 19:46:50.393406016 -0700
@@ -0,0 +1,21 @@
+<?php
+
+// Located in discussion.php
+// After comment-author
+function thematic_commentmeta() {
+  $content = '<div class="comment-meta">' .
+    sprintf( __('Posted %1$s at %2$s <span class="meta-sep">|</span> <a href="%3$s" title="Permalink to this comment">Permalink</a>', 'thematic' ),
+           get_comment_date(),
+           get_comment_time(),
+           '#comment-' . get_comment_ID() ) .
+    ' <span class="meta-sep">|</span> ' .
+    sprintf('<span class="edit-link"><a class="comment-edit-link" href="%1$s" title="%2$s">%3$s</a></span>',
+            get_edit_comment_link(),
+            __( 'Edit comment' ),
+            __( 'Edit', 'thematic' ) ) .
+    '</div>\n';
+
+  echo apply_filters('thematic_commentmeta', $content);
+} // end thematic_commentmeta
+
+?>
diff -uNr -x .svn thematic-orig/library/extensions/discussion.php thematic-comment-meta-filter/library/extensions/discussion.php
--- thematic-orig/library/extensions/discussion.php	2010-05-30 18:54:18.457303739 -0700
+++ thematic-comment-meta-filter/library/extensions/discussion.php	2010-05-30 19:23:24.597346128 -0700
@@ -7,11 +7,7 @@
     ?>
     	<li id="comment-<?php comment_ID() ?>" class="<?php thematic_comment_class() ?>">
     		<div class="comment-author vcard"><?php thematic_commenter_link() ?></div>
-    		<div class="comment-meta"><?php printf(__('Posted %1$s at %2$s <span class="meta-sep">|</span> <a href="%3$s" title="Permalink to this comment">Permalink</a>', 'thematic'),
-    					get_comment_date(),
-    					get_comment_time(),
-    					'#comment-' . get_comment_ID() );
-    					edit_comment_link(__('Edit', 'thematic'), ' <span class="meta-sep">|</span> <span class="edit-link">', '</span>'); ?></div>
+    		<?php thematic_commentmeta(); ?>
     <?php if ($comment->comment_approved == '0') _e("\t\t\t\t\t<span class='unapproved'>Your comment is awaiting moderation.</span>\n", 'thematic') ?>
             <div class="comment-content">
         		<?php comment_text() ?>
