diff -uNr -x .svn thematic-orig/library/extensions/comments-extensions.php thematic-per-comment-hooks/library/extensions/comments-extensions.php
--- thematic-orig/library/extensions/comments-extensions.php	2010-05-30 18:54:18.457303739 -0700
+++ thematic-per-comment-hooks/library/extensions/comments-extensions.php	2010-05-31 16:07:13.117370419 -0700
@@ -22,6 +22,20 @@
 }
 
 
+// Located in discussion.php
+// At the beginning of li#comment-[id]. Note that this is *per comment*.
+function thematic_abovecomment() {
+    do_action('thematic_abovecomment');
+}
+
+
+// Located in discussion.php
+// Just after the comment reply link. Note that this is *per comment*.
+function thematic_belowcomment() {
+    do_action('thematic_belowcomment');
+}
+
+
 // Located in comments.php
 // Just before #trackbacks-list
 function thematic_abovetrackbackslist() {
diff -uNr -x .svn thematic-orig/library/extensions/discussion.php thematic-per-comment-hooks/library/extensions/discussion.php
--- thematic-orig/library/extensions/discussion.php	2010-05-30 18:54:18.457303739 -0700
+++ thematic-per-comment-hooks/library/extensions/discussion.php	2010-05-31 16:23:25.213313871 -0700
@@ -6,6 +6,7 @@
 	$GLOBALS['comment_depth'] = $depth;
     ?>
     	<li id="comment-<?php comment_ID() ?>" class="<?php thematic_comment_class() ?>">
+    		<?php thematic_abovecomment() ?>
     		<div class="comment-author vcard"><?php thematic_commenter_link() ?></div>
     		<div class="comment-meta"><?php printf(__('Posted %1$s at %2$s <span class="meta-sep">|</span> <a href="%3$s" title="Permalink to this comment">Permalink</a>', 'thematic'),
     					get_comment_date(),
@@ -27,6 +28,7 @@
 					)));
 				endif;
 			?>
+    		<?php thematic_belowcomment() ?>
 <?php }
 
 // Custom callback to list pings in the Thematic style
